pkg/driver/bigquery/bigquery_test.go

```diff
@@ -0,0 +1,252 @@
+package bigquery
+
+import (
+	"database/sql"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/amacneil/dbmate/v2/pkg/dbmate"
+	"github.com/amacneil/dbmate/v2/pkg/dbutil"
+)
+
+func getURL() string {
+	return os.Getenv("BIGQUERY_TEST_URL")
+}
+
+func testBigQueryDriver(t *testing.T) *Driver {
+	u := dbutil.MustParseURL(getURL())
+	drv, err := dbmate.New(u).Driver()
+	require.NoError(t, err)
+
+	return drv.(*Driver)
+}
+
+func prepTestBigQueryDB(t *testing.T) *sql.DB {
+	drv := testBigQueryDriver(t)
+
+	// drop any existing database
+	err := drv.DropDatabase()
+	require.NoError(t, err)
+
+	// create database
+	err = drv.CreateDatabase()
+	require.NoError(t, err)
+
+	// connect database
+	db, err := drv.Open()
+	require.NoError(t, err)
+
+	return db
+}
+
+func TestGetDriver(t *testing.T) {
+	db := dbmate.New(dbutil.MustParseURL("bigquery://test/db_mate"))
+	drvInterface, err := db.Driver()
+	require.NoError(t, err)
+
+	// driver should have URL and default migrations table set
+	drv, ok := drvInterface.(*Driver)
+	require.True(t, ok)
+	require.Equal(t, db.DatabaseURL.String(), drv.databaseURL.String())
+	require.Equal(t, "schema_migrations", drv.migrationsTableName)
+}
+
+func TestConnectionString(t *testing.T) {
+	cases := []struct {
+		input    string
+		expected string
+	}{
+		{"bigquery://projectid/dataset", "bigquery://projectid/dataset?disable_auth=false"},
+		{"bigquery://projectid/location/dataset", "bigquery://projectid/location/dataset?disable_auth=false"},
```

Comment:
What is this `/location` portion in the URL path?  Does the underlying `go-gorm/bigquery` driver understand this?

Reply:
https://github.com/go-gorm/bigquery/blob/master/driver/driver.go#L104-L106 - Yes it does understand `location`. Location path is to specify location for the BigQuery Dataset which is optional.

Reply:
Maybe I'm mistaken, but I don't think that code does what you think it does.  Look at this example:

https://go.dev/play/p/hfmEYPPbhm9

```
func main() {
	u, _ := url.Parse("bigquery://projectid/location/dataset")
	fmt.Printf("u = %#v\n", u)

	fields := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
	fmt.Printf("len(fields) = %d\n", len(fields))
	for i, v := range fields {
		fmt.Printf("fields[%d] = %#v\n", i, v)
	}

	if len(fields) == 2 {
		fmt.Printf("config.location = fields[0] = %#v\n", fields[0])
	}
}
```

Notice how `config.location` will be set to `location` and not `dataset`.  That `go-gorm/bigquery` driver has a number of subtle bugs, because it accomodates improperly formed URLs, where they expect the project ID in the `host` portion of the URL, and the dataset ID in the `path` portion of the URL.